docs/preview/features/secret-store/provider/hashicorp-vault.md

```diff
@@ -130,4 +130,114 @@ public class Program
 }
 ```
 
+### Custom implementation
```

Comment:
All looks good to me, I just don't know if we have to add all of this documentation, what do you think @fgheysels ?

Reply:
We can maybe minimize it. Only show them in the overloads but not a fully working example like now. Would that be better?

Reply:
I think it's good to have thorough documentation :)
Drawback is offcourse that you also need to maintain it, but I'm a fan of this kind of docs.   It's clear and explains what must be done.

It's also a good idea to mention the fact that you can create a custom extension method for using your custom type, but, putting the example code for that might be a bit too much.   You could also refer to a source file which defines this extension method for the providers that we provide.

Reply:
Sounds good, let's keep it as is